cime_config/machines/config_machines.xml

```diff
@@ -362,8 +362,9 @@
         <arg name="label"> --label</arg>
         <arg name="num_tasks"> -n {{ total_tasks }} -N {{ num_nodes }}</arg>
         <arg name="thread_count">-c $SHELL{echo 128/`./xmlquery --value MAX_MPITASKS_PER_NODE`|bc}</arg>
-        <arg name="binding"> $SHELL{if [ 64 -ge `./xmlquery --value MAX_MPITASKS_PER_NODE` ]; then echo "--cpu_bind=cores"; else echo "--cpu_bind=threads";fi;} </arg>
+        <arg name="binding"> $SHELL{if [ 64 -ge `./xmlquery --value MAX_MPITASKS_PER_NODE` ]; then echo "--cpu-bind=cores"; else echo "--cpu-bind=threads";fi;} </arg>
         <arg name="placement"> -m plane=$SHELL{echo `./xmlquery --value MAX_MPITASKS_PER_NODE`}</arg>
+        <arg name="gpu-bind"> /global/cfs/cdirs/e3sm/tools/set_affinity_npergpu.sh $SHELL{echo `./xmlquery --value MAX_MPITASKS_PER_NODE`}</arg>
```

Comment:
What does this set_affinity_npergu.sh file do?

Reply:
Oh, I didn't realize that `set_affinity_npergu.sh` is getting added to the `srun` something like this:

```
srun  --label  -n 512 -N 8 -c 2  --cpu-bind=cores   -m plane=64  /global/cfs/cdirs/e3sm/tools/set_affinity_npergpu.sh 64 /pscratch/sd/g/gbisht/e3sm_scratch/pm-gpu/Dlnd.pm-gpu.gpu.NorthAmerica1km.GRFR.aa57ca56c8.NTASKS_32.gnugpu.2025-12-05/bld/e3sm.exe >> e3sm.log.$LID 2>&1
```